staging/src/k8s.io/apiserver/pkg/registry/generic/registry/store.go

```diff
@@ -445,6 +445,7 @@ func (e *Store) WaitForInitialized(ctx genericapirequest.Context, obj runtime.Ob
 				}
 			}
 		case <-ctx.Done():
+			return nil, ctx.Err()
```

Comment:
This seems like a bug to me. Without this `return` the loop becomes a hot loop once the context is done.